### ISSUE TITLE
[vLLM] Idle rows get a spurious token 0 ('!') appended during chunked prefill

### ISSUE THREAD

**Issue by ssaliceTT** [tenstorrent/tt-xla#5830]
### Describe the bug

In a mixed length batch, a request that has finished prefill sits idle while other requests work through their remaining prefill chunks. Each idle step appends a bogus token to that request's output, which decodes as `!`. Long outputs look like `!!!!`.

Chain:

1. `logits_indices[row] = num_scheduled_tokens[row] - 1` (`model_runner.py:1613`). A row with 0 scheduled tokens gets `-1`.
2. `select_hidden_states` gathers `hidden_states[row, -1, :]`, which is the last padded position, so all zeros.
3. Zero hidden state gives uniform logits, so `argmax` returns index 0. Token id 0 decodes to `!` in the Qwen tokenizer.
4. The discard only covers partial prefills (`model_runner.py:2410`):

```python
if seq_len >= req_state.num_tokens:
    request_seq_lens.append((i, req_state, seq_len))   # token accepted
else:
    # Ignore the sampled token from the partial request.
```

An idle row has already finished prefill, so `seq_len == num_tokens` and the token is accepted and appended.

Needs a mixed length batch. If every prompt finishes prefill in the same step, no row idles. Under data parallelism the effect is broader, because finished requests are deliberately retained as inert rows to preserve slot to replica affinity.

Observed on an 8x4 galaxy with Qwen3-32B, 16 prompts (14 short, 2 long). The short prompt in row 0 returned `!!!!`. The same slot was coherent in an earlier run where it held a long prompt, so it tracks whether the row idles, not the prompt.

### Steps to reproduce the issue

1. Enable chunked prefill, for example `additional_config={"prefill_chunk_size": 128}` with `max_model_len=1024`.
2. Submit a batch mixing prompts longer than the chunk size with prompts shorter than it.
3. Generate greedily.
4. Inspect `output.token_ids` for the short prompts. A leading `0` (which decodes to `!`) is the signature.

### Expected behavior

A row that scheduled no tokens produced no real logits, so its sampled token should be discarded rather than appended. Output should contain only tokens the model actually generated.

### Please complete the following environment information

- Ubuntu 24.04, Python 3.12
- vLLM 0.22.1
- tt-xla built from source, main at 1ac35af3d
- Hardware: 8x4 galaxy (Qwen3-32B) and n300-llmbox (Qwen3-0.6B)

**Comment by ssaliceTT** [comment]
Seems fixed so closing
